app/src/main/java/com/example/pinterbest/LoginFragment.kt

```diff
@@ -24,30 +36,127 @@ class LoginFragment : Fragment() {
+        }
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
-    private fun isValidLogIn(): Boolean {
-        return Validator.isValidName(binding.usernameBox.text.toString().trim()) &&
-            Validator.isValidPassword(binding.passwordBox.text.toString().trim())
+    private fun setUpBottomNavigationItem() {
+        (activity as MainActivity).binding.bottomNavigation.menu
+            .getItem(MainActivity.HOME_POSITION_BNV).isChecked = true
+    }
+
+    private fun loginProvider(view: View, response: Response<ResponseBody>) {
+        when (response.code()) {
+            SUCCESS -> {
+                saveSession(response)
+                view.findNavController()
+                    .navigate(R.id.action_loginFragment_to_homeFragment)
+                setUpBottomNavigationItem()
+            }
+            INVALID_DATA ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Предоставлены неверные учетные данные",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            WRONG_PASSWORD ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Неправильный пароль",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ALREADY_AUTHORIZED ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Вы уже авторизованы. Сначала выйдите из системы",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            USER_NOT_FOUND ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Пользователь не найден",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+        }
+    }
+
+    private fun saveSession(response: Response<ResponseBody>) {
+        val regex =
+            """['session_id=']+(?<=session_id=).{56}""".toRegex()
+        val cookie = regex.find(response.headers().values("Set-Cookie")[0])
+        sessionManager.saveUserData(cookie?.value ?: "")
+    }
+
+    private fun getUserData(): UserLogin {
+        return UserLogin(
+            binding.usernameBox.text.toString(),
+            binding.passwordBox.text.toString()
+        )
+    }
+
+    private fun validateUserFields(): Boolean {
+        return Validator(ResourceProvider(resources))
+            .isValidName(binding.usernameBox, true) &&
+            Validator(ResourceProvider(resources))
+                .isValidPassword(binding.passwordBox, true)
+    }
+
+    companion object {
+        const val SUCCESS = 204
+        const val INVALID_DATA = 400
+        const val WRONG_PASSWORD = 401
+        const val ALREADY_AUTHORIZED = 403
+        const val USER_NOT_FOUND = 404
```

Comment:
this codes are also data layer entities as well as `Response`

Reply:
Yes